src/main-process/atom-window.coffee

```diff
@@ -40,9 +40,15 @@ class AtomWindow
     if process.platform is 'linux'
       options.icon = @constructor.iconPath
 
+    if @applyTitleBarSetting()
+      options.titleBarStyle = 'hidden'
+
     @browserWindow = new BrowserWindow options
     global.atomApplication.addWindow(this)
 
+    if @applyTitleBarSetting()
+      @browserWindow.setSheetOffset(23)
```

Comment:
Hey, good call. I think we might be able to read `offsetHeight` after the title bar has been rendered to account for CSS-induced height changes